src/Data/Group/Free/Product.hs

```diff
@@ -0,0 +1,69 @@
+-- |
+-- Module       : Data.Group.Free.Product
+-- Copyright    : (c) 2020 Reed Mullanix, Emily Pillmore
+-- License      : BSD-style
+--
+-- Maintainer   : Reed Mullanix <reedmullanix@gmail.com>,
+--                Emily Pillmore <emilypi@cohomolo.gy>
+--
+-- Stability    : stable
+-- Portability  : non-portable
+--
+-- This module provides definitions for the 'FreeProduct' of two groups,
+-- along with useful combinators.
+--
+module Data.Group.Free.Product
+( FreeProduct(..)
+, simplify
+, coproduct
+) where
+
+import Data.Bifunctor
+import Data.Group
+
+import Data.Sequence (Seq(..))
+import qualified Data.Sequence as Seq
+
+
+-- | The free product of two 'Group's.
+--
+-- __Note:__ This does nto perform simplification upon multiplication or construction.
+-- To do this, one should use 'simplify'.
+newtype FreeProduct g h = FreeProduct { runFreeProduct :: Seq (Either g h) }
+    deriving (Show, Eq, Ord)
```

Comment:
as noted in the comment above, the `Eq` and `Ord` instances for unsimplified words will be wrong. Equivalence should not be derived here